sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/LN0AdapterTest.java

```diff
@@ -343,34 +344,27 @@ void testFindMatch() throws Exception {
         assertFalse(ln0Adapter.findMatch(doTypeName2,daTypeName).isPresent());
     }
 
-    @Test
-    void testHasControlBlockAndAddControlBlock() {
-        LN0 tln = new LN0();
-        tln.getLnClass().add(TLLN0Enum.LLN_0.value());
-        LN0Adapter lnAdapter = new LN0Adapter(null,tln);
-
-        ReportControlBlock controlBlock = new ReportControlBlock();
-        controlBlock.setName("rpt");
-        controlBlock.setConfRev(2L);
-        assertDoesNotThrow(()->lnAdapter.addControlBlock(controlBlock));
-        assertTrue(lnAdapter.hasControlBlock(controlBlock));
-
-        GooseControlBlock gooseControlBlock = new GooseControlBlock();
-        gooseControlBlock.setName("gse");
-        gooseControlBlock.setId("g1");
-        assertDoesNotThrow(()->lnAdapter.addControlBlock(gooseControlBlock));
-        assertTrue(lnAdapter.hasControlBlock(gooseControlBlock));
-
-        SMVControlBlock smvControlBlock = new SMVControlBlock();
-        smvControlBlock.setName("smv");
-        smvControlBlock.setId("s1");
-        assertDoesNotThrow(()->lnAdapter.addControlBlock(smvControlBlock));
-        assertTrue(lnAdapter.hasControlBlock(smvControlBlock));
+    @ParameterizedTest
+    @EnumSource(value = ControlBlockEnum.class, mode = EnumSource.Mode.EXCLUDE, names = "LOG")
+    void hasControlBlock_should_return_true(ControlBlockEnum controlBlockEnum) {
```

Comment:
this 3 tests couldn't them be parameterized in one test?

Reply:
I don't understand, `hasControlBlock_should_return_true` is already a ParameterizedTest. Can you be more specific ?

Reply:
OK, as discussed, I created a parametrized test for `addControlBlock`